scripts/lint.sh

```diff
@@ -93,6 +93,29 @@ function test_require_equal_len {
   fi
 }
 
+function test_require_nil {
+  if grep -R -o -P 'require\..+?!= nil' .; then
+    echo ""
+    echo "Use require.NotNil when testing for nil inequality."
+    echo ""
+    return 1
+  fi
+
+  if grep -R -o -P 'require\..+?== nil' .; then
+    echo ""
+    echo "Use require.Nil when testing for nil equality."
+    echo ""
+    return 1
+  fi
+
+  if grep -R -zo -P 'require\.ErrorIs.+?nil\)\n' .; then
```

Comment:
Why's there a newline at the end of this?

Reply:
In case of `require.ErrorIs(a.SomeFunc(nil), someErr)`. Now that https://github.com/ava-labs/avalanchego/pull/1486 is in though, that `\n` is extra - removed.

```suggestion
  if grep -R -o -P 'require\.ErrorIs.+?nil\)' .; then
```